server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryFactory.java

```diff
@@ -37,17 +39,25 @@ public RemoteSegmentStoreDirectoryFactory(Supplier<RepositoriesService> reposito
 
     @Override
     public Directory newDirectory(String repositoryName, IndexSettings indexSettings, ShardPath path) throws IOException {
+        String indexUUID = indexSettings.getIndex().getUUID();
+        String shardId = String.valueOf(path.getShardId().getId());
+        return newDirectory(repositoryName, indexUUID, shardId);
+    }
+
+    public Directory newDirectory(String repositoryName, String indexUUID, String shardId) throws IOException {
         try (Repository repository = repositoriesService.get().repository(repositoryName)) {
```

Comment:
this looks like a bug where `close` will be called on the `Repository` being used here ?